File: Data Structures & Algorithms/anagram-groups/submission-6.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        store = defaultdict(list)
        for string in strs:
            count = [0]*26
            for letter in string:
                count[ord(letter)-ord('a')] += 1
            store[tuple(count)].append(string)
        res = []
        for item in store.values():
            tmp = []
            for word in item:
                tmp.append(word)
            res.append(tmp)
        print(store)
        return(res)

File: Data Structures & Algorithms/top-k-elements-in-list/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let store = new Map();
        for(let i = 0; i < nums.length; i++ ) {
            store.set(nums[i], (store.get(nums[i]) || 0) + 1)
        }
        let res = [];
        let counter = 0;
        let tmp = [];
        for(const [number, count] of store) {
            tmp.push([number,count]);  
        }
        // console.log(tmp);
        tmp.sort((a,b) => b[1]-a[1]);
        for(let top of tmp) {
            res.push(top[0]);
            counter += 1
            if (counter == k) return res;
        }
    }
}
